### PR TITLE
Make IDL tests actually run (closes #7270)

### LINKED ISSUE #7069: PaymentRequest interfaces should be fetch()`ed
The PaymentRequest interface tests are currently copy/pasted from the spec... but this means that they risk falling out of date. Instead, we should `fetch()` them.

### LINKED ISSUE #7270: https://w3c-test.org/payment-request/interfaces.https.html is broken
seems to be some error loading JavaScript libraries that do not exist in the file path.

//CC @marcoscaceres

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #7069, #7270

### CODE REVIEW

**Comment by marcoscaceres** [comment]
cc AmazingJaze

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18245)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/272742345)

**Comment by marcoscaceres** [comment]
@zcorpan, would appreciate your review.

**Comment by marcoscaceres** [comment]
There might be a bug in the idlharness because it's reporting `.toJSON()` as failing in both Chrome and Firefox... or it might be both browsers are buggy :) But seriously, who could we ping to look into that?

**Comment by domenic** [comment]
Yes, toJSON is broken per https://github.com/w3c/web-platform-tests/issues/6974 unfortunately.

File: payment-request/interfaces.https.html [hunk 1]
```diff
@@ -5,136 +5,26 @@
...
+  );
+  idlArray.add_idls(testedIdl);
+  const methods = "[{supportedMethods: 'foo'}]";
+  const amount = "{currency: 'USD', value: '0'}";
+  const details = `{total: {label: 'bar', amount: ${amount} }}`;
+  idlArray.add_objects({
+    PaymentRequest: [`new PaymentRequest(${methods}, ${details})`],
+  });
+  idlArray.test();
+}, "Must have a conforming IDL implementation of the Payment Request API.");
```
Comment (domenic) [inline payment-request/interfaces.https.html]: I don't think if this test passes that will be true. The test will just pass if the fetch succeeded, etc. So I would label this "Setup for Payment Request API IDL tests" or similar.

File: payment-request/interfaces.https.html [hunk 2]
```diff
@@ -5,136 +5,26 @@
...
-    idlArray.add_idls(node.textContent);
-  }
-});
-idlArray.add_objects({
-  PaymentRequest: ["new PaymentRequest([{supportedMethods: 'foo'}], {total: {label: 'bar', amount: {currency: 'BAZ', value: '0'}}})"]
-});
-idlArray.test();
+promise_test(async () => {
+  const idlArray = new IdlArray();
+  const untestedIDL = `
```
Comment (zcorpan) [inline payment-request/interfaces.https.html]: Maybe better to fetch all of interfaces/dom.idl ?

Comment (marcoscaceres) [reply payment-request/interfaces.https.html]: Yeah, why not! 💃

**Comment by marcoscaceres** [comment]
Ok, as per @zcorpan this now relies on dom.idl. Important to note: DOM includes the new `AbortController` and `AbortSignal` interfaces, and it's unclear when those will land in Edge. 

@AmazingJaze, @MSFTkihans, you can ignore those failures - also, you will see `.toJSON()` fail. That's a bug in the test harness, which should hopefully be fixed soon.

File: payment-request/interfaces.https.html [hunk 3]
```diff
@@ -5,136 +5,27 @@
...
+  for(const url of idlURLs){
+    const idlText = await fetch(url).then(r => r.text());
+    idlArray.add_idls(idlText);
   }
-});
-idlArray.add_objects({
-  PaymentRequest: ["new PaymentRequest([{supportedMethods: 'foo'}], {total: {label: 'bar', amount: {currency: 'BAZ', value: '0'}}})"]
-});
-idlArray.test();
+  // typedef EventHandler from HTML
```
Comment (marcoscaceres) [inline payment-request/interfaces.https.html]: Filed a bug on DOM spec regarding having to do this:
https://github.com/whatwg/dom/issues/509

Let's see what @annevk thinks.

Comment (marcoscaceres) [reply payment-request/interfaces.https.html]: Anne suggested maybe splitting out EventHandler into it's own IDL file. That might make sense given how many specs are relying on it (but where it wouldn't make sense to import all of HTML). 

See: 

```
git grep "interface EventHandler"
```

Comment (marcoscaceres) [reply payment-request/interfaces.https.html]: (can do this as a separate bug, if we agree to that)

Comment (marcoscaceres) [reply payment-request/interfaces.https.html]: Proposal: https://github.com/w3c/web-platform-tests/pull/7331

**Comment by marcoscaceres** [comment]
@zcorpan, bump.

**Comment by marcoscaceres** [comment]
Yay! .toJSON() is fixed!!! 💖 https://github.com/w3c/web-platform-tests/pull/7308

**Review by zcorpan — APPROVED** [review]

**Pull request merged**